worlds/yoshisisland/Items.py

```diff
@@ -0,0 +1,124 @@
+from typing import Dict, Set, Tuple, NamedTuple, Optional
+
+class ItemData(NamedTuple):
+    category: str
+    code: int
+    amount: Optional[int] = 1
+    progression: bool = False

```

Comment:
given most of your items are progression it probably makes more sense to make that the default.